google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemTest.java

```diff
@@ -198,24 +201,26 @@ public void testListFiles() throws IOException {
       List<Path> paths = new ArrayList<>();
       goodPaths.add(fs.getPath("dir/angel"));
       goodPaths.add(fs.getPath("dir/alone"));
-      paths.add(fs.getPath("dir/dir2/another_angel"));
+      goodPaths.add(fs.getPath("dir/dir2/angel"));
+      paths.add(fs.getPath("dir1/another_angel"));
       paths.add(fs.getPath("atroot"));
       paths.addAll(goodPaths);
-      goodPaths.add(fs.getPath("dir/dir2/"));
       for (Path path : paths) {
         Files.write(path, ALONE.getBytes(UTF_8));
       }
+      assertThat(goodPaths.size()).isEqualTo(3);
+      assertThat(paths.size()).isEqualTo(5);
 
       List<Path> got = new ArrayList<>();
-      for (Path path : Files.newDirectoryStream(fs.getPath("/dir/"))) {
-        got.add(path);
+      for (Blob blob : STORAGE.list("bucket", Storage.BlobListOption.prefix("/dir/")).getValues()) {
+        got.add(fs.getPath(blob.getName()));
       }
       assertThat(got).containsExactlyElementsIn(goodPaths);
 
       // Must also work with relative path
       got.clear();
-      for (Path path : Files.newDirectoryStream(fs.getPath("dir/"))) {
-        got.add(path);
+      for (Blob blob : STORAGE.list("bucket", Storage.BlobListOption.prefix("dir/")).getValues()) {
+        got.add(fs.getPath(blob.getName()));
       }
       assertThat(got).containsExactlyElementsIn(goodPaths);
```

Comment:
I'm not sure that it's a right thing to use Storage directly in NIO tests.
When  test did:
```
for (Path path : Files.newDirectoryStream(fs.getPath("dir/"))) {
        got.add(path);
```
test verified nio.
Now test verifies Storage, not nio.

Reply:
@dmitry-fa update the test and removed the use of storage